service-templates/apigw-lambda-svc/v1/pipeline_infrastructure/main.tf

```diff
@@ -0,0 +1,393 @@
+  policy = data.aws_iam_policy_document.publish_role_policy_document.json
+}
+
+resource "aws_iam_role_policy_attachment" "publish_role_policy_attachment" {
+  policy_arn = aws_iam_policy.publish_role_policy.arn
+  role       = aws_iam_role.publish_role.name
+}
+
+resource "aws_iam_role" "deployment_role" {
+  name_prefix = "deployment-role"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "codebuild.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "deployment_role_policy" {
+  policy = data.aws_iam_policy_document.deployment_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "deployment_role_policy_attachment" {
+  policy_arn = aws_iam_policy.deployment_role_policy.arn
+  role       = aws_iam_role.deployment_role.name
+}
+
+resource "aws_s3_bucket" "pipeline_artifacts_bucket" {
+  bucket_prefix = "pipeline-artifacts-bucket"
+}
+
+resource "aws_s3_bucket_public_access_block" "pipeline_artifacts_bucket_access_block" {
+  bucket                  = aws_s3_bucket.pipeline_artifacts_bucket.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_server_side_encryption_configuration" "pipeline_artifacts_bucket_encryption" {
+  bucket = aws_s3_bucket.pipeline_artifacts_bucket.id
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm     = "aws:kms"
+      kms_master_key_id = aws_kms_key.pipeline_artifacts_bucket_key.arn
+    }
+  }
+}
+
+resource "aws_kms_key" "pipeline_artifacts_bucket_key" {
+  policy = data.aws_iam_policy_document.pipeline_artifacts_bucket_key_policy.json
+}
+
+resource "aws_kms_alias" "pipeline_artifacts_bucket_key_alias" {
+  target_key_id = aws_kms_key.pipeline_artifacts_bucket_key.id
+  name          = "alias/codepipeline-encryption-key-${var.service.name}"
+}
+
+resource "aws_iam_role" "pipeline_role" {
+  name_prefix = "pipeline-role"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "codepipeline.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "pipeline_role_policy" {
+  policy = data.aws_iam_policy_document.pipeline_role_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "pipeline_role_policy_attachment" {
+  policy_arn = aws_iam_policy.pipeline_role_policy.arn
+  role       = aws_iam_role.pipeline_role.name
+}
+
+resource "aws_codepipeline" "pipeline" {
+  name     = "${var.service.name}-pipeline"
+  role_arn = aws_iam_role.pipeline_role.arn
+
+  stage {
+    name = "Source"
+    action {
+      category  = "Source"
+      name      = "Source"
```

Comment:
This should be Source -> [Checkout](https://github.com/syedsaadahmed1/aws-proton-sample-templates/blob/main/service-templates/apigw-lambda-svc/v1/pipeline_infrastructure/cloudformation.yaml#L559)